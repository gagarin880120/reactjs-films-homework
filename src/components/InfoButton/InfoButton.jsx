import React from 'react';
import styles from './InfoButton.module.scss';

export default function InfoButton(props) {
  return (
    <button
      onClick={props.onInfoButtonClick}
      className={styles.infoButton}
      data-testid={props.testid}
    >
      View Info
    </button>
  )
}
