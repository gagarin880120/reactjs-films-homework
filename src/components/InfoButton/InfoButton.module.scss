.infoButton {
  color: #fff;
  font-size: 16px;
  outline: 0;
  background: transparent;
  border: 1px solid #fff;
  border-radius: 3px;
  width: 114px;
  height: 35px;
  cursor: pointer;
}

.infoButton:active {
  transform: translateY(2px);
}

@media screen and (max-width: 600px) {
  .infoButton {
    width: 70px;
    height: 45px;
  }
}
