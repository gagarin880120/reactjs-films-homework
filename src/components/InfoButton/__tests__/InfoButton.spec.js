import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import InfoButton from '../InfoButton';

describe('InfoButton component', () => {
  test('should render without crashing', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<InfoButton />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
