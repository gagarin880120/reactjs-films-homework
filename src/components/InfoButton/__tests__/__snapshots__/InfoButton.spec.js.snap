// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoButton component should render without crashing 1`] = `
<button
  className="infoButton"
  onClick={null}
  type="button"
>
  View Info
</button>
`;
